plugin/storage/badger/spanstore/writer.go

```diff
@@ -179,3 +209,51 @@ func createTraceKV(span *model.Span, encodingType byte, startTime uint64) ([]byt
 
 	return key, bb, err
 }
+
+// This method is only for testing purpose to test backward compatibility, once dual-lookup is removed, this method can be removed
+func (w *SpanWriter) writeSpanWithOldIndex(span *model.Span) error {
```

Comment:
is this supposed to be the old way of writing? Why create a new method then instead of having a new method only for NEW way. That way the diff will be smaller and more readable.